matcher/solvers/randomized_solver.py

```diff
@@ -169,13 +168,29 @@ def solve(self):
         self.logger.debug('start fractional_assignment_solver')
 
         self.expected_cost = 0
+        self.logger.debug('call to Solve()')
```

Comment:
@sjecmen I added more logs and the line that is hanging is the call to the ortools Solve()

```
2021-06-09 00:07:10,522 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:76] Thread-1 Checking graph inputs
2021-06-09 00:07:10,537 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:104] Thread-1 Finished checking graph inputs
2021-06-09 00:07:10,537 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:124] Thread-1 construct_solver
2021-06-09 00:07:58,139 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:156] Thread-1 Finished construct_solver
2021-06-09 00:07:58,663 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:124] Thread-1 construct_solver
2021-06-09 00:08:44,282 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:156] Thread-1 Finished construct_solver
2021-06-09 00:08:44,823 DEBUG: [in /home/openreview/openreview-matcher/matcher/core.py:155] Thread-1 Solving solver
2021-06-09 00:08:44,823 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:161] Thread-1 solve
2021-06-09 00:08:44,823 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:109] Thread-1 Checking if demand is in range
2021-06-09 00:08:44,823 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:115] Thread-1 Total demand is (9846), min review supply is (8384), and max review supply is (10448)
2021-06-09 00:08:44,823 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:120] Thread-1 Finished checking if demand is in range
2021-06-09 00:08:44,823 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:168] Thread-1 start fractional_assignment_solver
2021-06-09 00:08:44,823 DEBUG: [in /home/openreview/openreview-matcher/matcher/solvers/randomized_solver.py:171] Thread-1 call to Solve()
2021-06-09 00:10:04,015 DEBUG: [in /home/openreview/openreview-matcher/matcher/service/__init__.py:31] MainThread Starting app
```

I'm not sure how to debug this or is there a way to run ortools in debug mode?

Reply:
Maybe the ORTools LP solver is just too slow for a problem this large? I don't think that scaling up the constraints should have made it any slower (and it seems like this issue was occurring before I made that change anyway). 

Now that I've made all the constraints integral, I can change to using the ORTools min-cost flow solver like the other assignment algorithms use. There's also a way to add a time limit to the LP solver, but I don't think this would help solve anything.

Reply:
If you check the last two lines of the the logs, the server hangs after 1 min and 20 seconds of calling the Solver(). It is weird that no error is thrown.

Reply:
I agree it's strange, I'm not sure what's going on there. I don't see any debug mode for the solver so I'm also unsure how to debug it further. I'm working on using the min-cost flow solver from MinMaxSolver at the moment, but it's taking some time.